src/createServer.js

```diff
@@ -1,3 +1,64 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const http = require('http');
+const { convertToCase } = require('./convertToCase');
+
+function createServer() {
+  const server = http.createServer((req, res) => {
+    res.setHeader('Content-Type', 'application/json');
+
+    const normaliseUrl = new URL(req.url, `http://${req.headers.host}`);
+    const { pathname, searchParams } = normaliseUrl;
+    const text = pathname.slice(1);
+    const params = new URLSearchParams(searchParams);
+    const toCase = params.get('toCase');
+    const cases = ['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER'];
+    const errors = [];
+
+    if (text === '') {
+      errors.push({
+        message: 'Text to convert is required.'
+        + ' Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+      });
+    }
+
+    if (!toCase) {
+      errors.push({
+        message: '"toCase" query param is required.'
+        + ' Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+      });
+    }
+
+    if (!cases.includes(toCase) && toCase) {
+      errors.push({
+        message: 'This case is not supported.'
+        + ' Available cases: SNAKE, KEBAB, CAMEL, PASCAL, UPPER.',
+      });
+    }
+
+    if (errors.length !== 0) {
```

Comment:
if (errors.length) {
  ...
}

Reply:
Thank you very much for the commits)  I fixed everything and then some